src/Jwt/Command/JwtGeneratorCommand.php

```diff
@@ -62,23 +63,20 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
         $user = reset($users);
 
-        if ($input->hasOption('role')
+        if ($input->hasOption('email')
+            && null !== $input->getOption('email'))
+        {
+            $user = $this->findByEmail($input->getOption('email'), $output, $user, $users);
+        }
+        else if ($input->hasOption('role')
             && null !== $input->getOption('role')
-            && !in_array($input->getOption('role'), $user->getRoles())
-        ) {
-            $find = false;
-            foreach ($users as $item) {
-                if (in_array($input->getOption('role'), $item->getRoles())) {
-                    $user = $item;
-                    $find = true;
-                }
-            }
-
-            if (!$find) {
-                $output->writeln(sprintf('<error>User with role "%s" doesn\'t exist.<error>', $input->getOption('role')));
+            && !in_array($input->getOption('role'), $user->getRoles()))
+        {
+            $user = $this->findOneByRole($input, $output, $users);
+        }
```

Comment:
If we specify email and role, role will be ignored. This is due to refactoring in splitted methods. You probably should consider to add an error message or drop this refactoring and use a single (fat, I admit) condition.

Reply:
I'm assuming that if you use the 'email' flag you must know which user you want and therefore you need to know which role this user have